test/helpers/doLogin.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+const env = require('../environment');
+const b64 = require('./b64');
+const logger = require('../helpers/logger');
+const timing = require('../helpers/timing');
+const LocalStorage = require('../pageobjects/localStorage');
+const LoginPage = require('../pageobjects/login');
+const CatalogPage = require('../pageobjects/catalog');
+
+const [favicon, mainCSS, configJS, html404] = [
+  `${env.consoleUrl}/images/favicon.png`,
+  `${env.consoleUrl}/styles/main.css`,
+  `${env.consoleUrl}/config.js`,
+  `${env.consoleUrl}/404.html`
+];
+
+// NOTE: this uses the default login form, it may not be available for
+// all environments
+const withOauthForm = (name = env.user.name, pass = env.user.pass) => {
+  browser.get(html404);
+  logger.setLogLevel('INFO');
+  const login = new LoginPage();
+  login.visit();
+  browser.driver.sleep(timing.oauthRedirect);
+  login.login({name, pass});
+  return browser.sleep(10 * 1000);
+};
+
+
+const withTokenInjectedIntoLocalStorage = (token) => {
+  // Load a page on same domain w/o an oauth redirect
+  browser.get(html404);
+  logger.setLogLevel('INFO');
+
+  const keys = {
+    token: 'LocalStorageUserStore.token',
+    tokenttl: 'LocalStorageUserStore.token.ttl',
+    user: 'LocalStorageUserStore.user',
+    userttl: 'LocalStorageUserStore.user.ttl'
+  };
+
+  const vals = {
+    token: token || env.auth.token,
+    tokenttl: '1523022514453',
```

Comment:
@jhadvig @spadgett 

This is currently a problem. I can pass an oauth token, but if the TTL is off, we still fail.  So, i'm sure that a Service Accout Token w/o a correct TTL will also fail.  Fishing around for a solution here.

If i use a fresh token & copy out ttl times, works fine.